packages/snaps-jest/src/matchers.ts

```diff
@@ -158,32 +158,87 @@ export const toRespondWithError: MatcherFunction<[expected: Json]> = function (
  * is intended to be used with the `expect` global.
  *
  * @param actual - The actual response.
- * @param expected - The expected notification message.
- * @param type - The expected notification type.
+ * @param expectedMessage - The expected notification message.
+ * @param expectedType - The expected notification type.
+ * @param expectedTitle - The expected notification title.
+ * @param expectedFooterLink - The expected footer link object.
  * @returns The status and message.
  */
 export const toSendNotification: MatcherFunction<
-  [expected: string, type?: EnumToUnion<NotificationType> | undefined]
-> = function (actual, expected, type) {
+  [
+    expectedMessage: string,
+    expectedType?: EnumToUnion<NotificationType> | undefined,
+    expectedTitle?: string | undefined,
+    expectedFooterLink?: { text: string; href: string } | undefined,
+  ]
+> = function (
+  actual,
+  expectedMessage,
+  expectedType,
+  expectedTitle,
+  expectedFooterLink,
+) {
   assertActualIsSnapResponse(actual, 'toSendNotification');
 
+  const notificationValidator = (
+    notification: SnapResponse['notifications'][number],
+  ) => {
+    const { type, message, title, content, footerLink } = notification;
+    const hasExpectedType = expectedType !== undefined;
+    const hasExpectedTitle = expectedTitle !== undefined;
+    const hasExpectedFooterLink = expectedFooterLink !== undefined;
+
+    if (!this.equals(message, expectedMessage)) {
+      return false;
+    }
+
+    if (hasExpectedType && type !== expectedType) {
+      return false;
+    }
+
+    if (type === NotificationType.InApp) {
+      if (hasExpectedTitle && !this.equals(title, expectedTitle)) {
+        return false;
+      }
+
+      if (title && !content) {
+        return false;
+      }
+
+      if (
+        hasExpectedFooterLink &&
+        !this.equals(footerLink, expectedFooterLink)
+      ) {
+        return false;
+      }
+    }
+
+    return true;
+  };
+
   const { notifications } = actual;
-  const pass = notifications.some(
-    (notification) =>
-      this.equals(notification.message, expected) &&
-      (type === undefined || notification.type === type),
-  );
+  const pass = notifications.some(notificationValidator);
 
   const message = pass
     ? () =>
         `${this.utils.matcherHint('.not.toSendNotification')}\n\n` +
-        `Expected: ${this.utils.printExpected(expected)}\n` +
-        `Expected type: ${this.utils.printExpected(type)}\n` +
+        `Expected message: ${this.utils.printExpected(expectedMessage)}\n` +
+        `Expected type: ${this.utils.printExpected(expectedType)}\n` +
+        `Expected title: ${this.utils.printExpected(expectedTitle)}\n` +
+        `Expected footer link: ${this.utils.printExpected(
+          expectedFooterLink,
+        )}\n` +
+        `Expected content id type: ${this.utils.printExpected('String')}\n` +
```

Comment:
What's this?

Reply:
Realized that the notify method is creating an interface (expanded view) but not exposing the interface id, so the id can't be used inside `toSendNotification`. I was attempting to do some validation to make sure the response had a string content id. I'm now wondering if I can somehow intercept the middleware to return the original jsx content instead of the interface id so users can also validate the content inside of `toSendNotification`

Reply:
In the latest commit, [2cb571f](https://github.com/MetaMask/snaps/pull/2956/commits/2cb571fd0f7c0ae28446286cd74924cfdefc957e), I'm trying to get the actual jsx content now and match against, returning the jsx content in the "received" part of the message. The example snap's test pass regardless if I have the wrong content passed into `toSendNotification` so I'm not sure where I'm going wrong here, would appreciate a second look.

Reply:
Nevermind, got it to actually work.